albumentations/augmentations/transforms.py

```diff
@@ -3177,3 +3178,133 @@ def get_transform_init_args_names(self):
+                Clipped between 0 and grid unit size. Default: (0,0).
+            random_offset (boolean): weather to offset the grid randomly between 0 and grid unit size - hole size
+                If 'True', entered shift_x, shift_y are ignored and set randomly. Default: `False`.
+            fill_value (int): value for the dropped pixels. Default = 0
+        Targets:
+            image
+        Image types:
+            uint8, float32
+        References:
+            https://arxiv.org/pdf/2001.04086.pdf7
+    """
+
+    def __init__(
+        self,
+        ratio: float = 0.5,
+        unit_size_min: int = None,
+        unit_size_max: int = None,
+        holes_number_x: int = None,
+        holes_number_y: int = None,
+        shift_x: int = 0,
+        shift_y: int = 0,
+        random_offset: bool = False,
+        fill_value: int = 0,
+        always_apply: bool = False,
+        p: float = 0.5,
+    ):
+        super(GridDropout, self).__init__(always_apply, p)
+        self.ratio = ratio
+        self.unit_size_min = unit_size_min
+        self.unit_size_max = unit_size_max
+        self.holes_number_x = holes_number_x
+        self.holes_number_y = holes_number_y
+        self.shift_x = shift_x
+        self.shift_y = shift_y
+        self.random_offset = random_offset
+        self.fill_value = fill_value
+        assert 0 < self.ratio < 1, "ratio must be between 0 and 1."
+
+    def apply(self, image, fill_value=0, holes=[], **params):
+        return F.cutout(image, holes, fill_value)
+
+    def get_params_dependent_on_targets(self, params):
+        img = params["image"]
+        height, width = img.shape[:2]
+        # set grid using unit size limits
+        if self.unit_size_min and self.unit_size_max:
+            assert (
+                2 <= self.unit_size_min <= self.unit_size_max
+            ), "Max unit size should be >= min size, both at least 2 pixels."
+            assert self.unit_size_max <= min(height, width), "Grid size limits must be within the shortest image edge."
+            self.unit_width = random.randint(self.unit_size_min, self.unit_size_max + 1)
+            self.unit_height = self.unit_width
+        else:
+            # set grid using holes numbers
+            if self.holes_number_x is None:
+                self.unit_width = max(2, width // 10)
+            else:
+                assert (
+                    1 <= self.holes_number_x <= width // 2
+                ), "The hole_number_x must be between 1 and image width//2."
+                self.unit_width = width // self.holes_number_x
+            if self.holes_number_y is None:
+                self.unit_height = max(min(self.unit_width, height), 2)
+            else:
+                assert (
+                    1 <= self.holes_number_y <= height // 2
+                ), "The hole_number_y must be between 1 and image height//2."
+                self.unit_height = height // self.holes_number_y
```

Comment:
Are you sure, that we need save `self.unit_width` and `self.unit_height` in this object scope?
I think we can use them as `unit_width` and `unit_height` without self, because for each image this values may be different. Also look to another variables like `hole_width` and `hole_height`